src/main/java/de/tum/in/www1/artemis/repository/ResultRepository.java

```diff
@@ -58,16 +58,19 @@ public interface ResultRepository extends JpaRepository<Result, Long> {
             FROM Result r
                 LEFT JOIN FETCH r.feedbacks f
                 LEFT JOIN FETCH f.testCase
-            WHERE r.completionDate =
-                (SELECT max(rr.completionDate) FROM Result rr
-                    WHERE rr.assessmentType = 'AUTOMATIC'
-                        AND rr.participation.exercise.id = :exerciseId
-                        AND rr.participation.student.id = r.participation.student.id
+                LEFT JOIN TREAT (r.participation AS ProgrammingExerciseStudentParticipation) sp
+            WHERE r.completionDate = (
+                    SELECT MAX(rr.completionDate)
```

Comment:
Maybe `r` could be `result` and `rr` could be `latest`

Reply:
latest wouldn't really make sense as it's not just one result but multiple. We just take the max of all completion date, not the result itself.

We can of course argue about/improve the var naming here but that's an overall issue and out of scope of these PRs.
I hope you can appreciate this. Thank you for your review though!